src/api/TKMember.m

```diff
@@ -218,6 +218,17 @@ - (void)addAliases:(NSArray<Alias *> *)toAddAliases
     NSMutableArray<MemberOperation *> *addAliasOps = [NSMutableArray array];
     NSMutableArray<MemberOperationMetadata *> *metadataArray = [NSMutableArray array];
     for (Alias *alias in toAddAliases) {
+        NSString *partnerId = retainedMember.partnerId;
+        if ([partnerId length] != 0) {
+            // Realm must equal member's partner ID if affiliated
+            if ([alias.realm length] != 0 && ![alias.realm isEqualToString:partnerId]) {
+                [NSException
+                   raise:NSInvalidArgumentException
+                   format:@"Invalid realm %@; expected %@", alias.realm, partnerId];
```

Comment:
In objc sdk, we should use:

    onError([NSError someInstance]);
    return;

to propagate the error instead of throwing the exception. More details in TkError.

Reply:
Done.